Grails 4 lost uuidConverter and jsr310DataBinding.

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

Here is a sample service:

```groovy
@Service(Example)
abstract class ExampleService {

    @Autowired
    @Qualifier('localDateTimeValueConverter')
    ValueConverter localDateTimeValueConverter

}
```

Grails 3 must add `grails-java8` plugin

```groovy
compile "org.grails.plugins:grails-java8"
```

It's workding in Grails 3.x, but failing in Grails 4:

```
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'dateTimeValueConverter': Unsatisfied dependency expressed through field 'localDateTimeValueConverter'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'grails.databinding.converters.ValueConverter' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true), @org.springframework.beans.factory.annotation.Qualifier(value=localDateTimeValueConverter)}
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:596)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:90)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:374)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1411)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:592)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:515)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:320)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:318)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:277)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1251)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1171)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:593)
	... 53 more
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'grails.databinding.converters.ValueConverter' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true), @org.springframework.beans.factory.annotation.Qualifier(value=localDateTimeValueConverter)}
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.raiseNoMatchingBeanFound(DefaultListableBeanFactory.java:1658)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1217)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1171)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:593)
	... 66 more
```

And I find the `localDateTimeValueConverter` has a `@Bean` annotation : 

https://github.com/grails/grails-core/blob/e45855b791e305c40e52bbd628699e428b0ff05a/grails-plugin-databinding/src/main/groovy/org/grails/databinding/converters/Jsr310ConvertersConfiguration.groovy#L136-L151

Also, here is the code in `Bootstrap.groovy` to test value converters:

```groovy
    GrailsWebDataBinder grailsWebDataBinder

    def init = { servletContext ->
        grailsWebDataBinder.conversionHelpers.each {
            println(it)
        }
    }
```

Output:

```
class [B=[org.grails.web.databinding.converters.ByteArrayMultipartFileValueConverter@6c2e9a45]
class java.lang.Long=[org.grails.databinding.converters.web.LocaleAwareNumberConverter@703749b2]
int=[org.grails.databinding.converters.web.LocaleAwareNumberConverter@495950ff]
class java.lang.Short=[org.grails.databinding.converters.web.LocaleAwareNumberConverter@5c2f6b29]
double=[org.grails.databinding.converters.web.LocaleAwareNumberConverter@2d8a76c0]
class java.util.TimeZone=[org.grails.databinding.converters.TimeZoneConverter@2193d5f8]
class java.util.Currency=[org.grails.databinding.converters.CurrencyValueConverter@7d1077fb]
class java.lang.Integer=[org.grails.databinding.converters.web.LocaleAwareNumberConverter@9b968d4]
class java.util.Date=[org.grails.databinding.converters.DateConversionHelper@350b02f9]
float=[org.grails.databinding.converters.web.LocaleAwareNumberConverter@5ac13d53]
class java.math.BigInteger=[org.grails.databinding.converters.web.LocaleAwareBigDecimalConverter@6c1c118b]
class java.lang.Double=[org.grails.databinding.converters.web.LocaleAwareNumberConverter@7567203f]
short=[org.grails.databinding.converters.web.LocaleAwareNumberConverter@31e3a338]
class java.math.BigDecimal=[org.grails.databinding.converters.web.LocaleAwareBigDecimalConverter@4ce207f9]
long=[org.grails.databinding.converters.web.LocaleAwareNumberConverter@ee6483a]
class java.lang.Float=[org.grails.databinding.converters.web.LocaleAwareNumberConverter@5ba6011b]
```

No `UUIDConverter` and Jsr310DataBinding, but Grails 3.3.10 works.

https://github.com/grails/grails-core/blob/e45855b791e305c40e52bbd628699e428b0ff05a/grails-plugin-databinding/src/main/groovy/org/grails/plugins/databinding/AbstractDataBindingGrailsPlugin.groovy#L79

https://github.com/grails/grails-core/blob/e45855b791e305c40e52bbd628699e428b0ff05a/grails-plugin-databinding/src/main/groovy/org/grails/plugins/databinding/AbstractDataBindingGrailsPlugin.groovy#L102-L104

I find the `uuidConverter(UUIDConverter)` and `jsr310DataBinding(Jsr310ConvertersConfiguration)` already add to beans, but cannot find in the `GrailsWebDataBinder`

How could I inject this bean ?

### Expected Behaviour

`localDateTimeValueConverter` should be injected.

### Actual Behaviour

UnsatisfiedDependencyException 

### Environment Information

- **Operating System**: Windows 10
- **Grails Version:** 4.0.0
- **JDK Version:** 1.8
- **Container Version (If Applicable):** TODO

--- a/grails-plugin-databinding/src/main/groovy/org/grails/databinding/converters/DefaultConvertersConfiguration.java
+++ b/grails-plugin-databinding/src/main/groovy/org/grails/databinding/converters/DefaultConvertersConfiguration.java
@@ -17,12 +17,13 @@
 package org.grails.databinding.converters;
 
 import grails.core.GrailsApplication;
+import grails.databinding.TypedStructuredBindingEditor;
+import grails.databinding.converters.FormattedValueConverter;
 import grails.databinding.converters.ValueConverter;
 import io.micronaut.context.exceptions.NoSuchBeanException;
 import org.grails.databinding.converters.web.LocaleAwareBigDecimalConverter;
 import org.grails.databinding.converters.web.LocaleAwareNumberConverter;
 import org.grails.plugins.databinding.DataBindingConfigurationProperties;
-import org.springframework.beans.BeansException;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.web.servlet.LocaleResolver;
@@ -38,14 +39,16 @@ public class DefaultConvertersConfiguration {
 
     private final DataBindingConfigurationProperties configurationProperties;
     private final LocaleResolver localResolver;
+    private final Jsr310ConvertersConfiguration jsr310ConvertersConfiguration;
 
     public DefaultConvertersConfiguration(GrailsApplication grailsApplication, DataBindingConfigurationProperties configurationProperties) {
         this.configurationProperties = configurationProperties;
+        jsr310ConvertersConfiguration = new Jsr310ConvertersConfiguration(configurationProperties);
         LocaleResolver localResolver;
         try {
             localResolver = grailsApplication.getMainContext().getBean(LocaleResolver.class);
         } catch (NoSuchBeanException e) {
-             localResolver = null;
+            localResolver = null;
         }
         this.localResolver = localResolver;
     }
@@ -64,6 +67,111 @@ public class DefaultConvertersConfiguration {
         converter.setTargetType(BigDecimal.class);
         converter.setLocaleResolver(localResolver);
         return converter;
+    }
+
+    @Bean("offsetDateTimeConverter")
+    FormattedValueConverter offsetDateTimeConverter() {
+        return jsr310ConvertersConfiguration.offsetDateTimeConverter();
+    }
+
+    @Bean("offsetDateTimeValueConverter")
+    ValueConverter offsetDateTimeValueConverter() {
+        return jsr310ConvertersConfiguration.offsetDateTimeValueConverter();
+    }
+
+    @Bean("offsetDateTimeStructuredBindingEditor")
+    TypedStructuredBindingEditor offsetDateTimeStructuredBindingEditor() {
+        return jsr310ConvertersConfiguration.offsetDateTimeStructuredBindingEditor();
+    }
+
+    @Bean("offsetTimeConverter")
+    FormattedValueConverter offsetTimeConverter() {
+        return jsr310ConvertersConfiguration.offsetTimeConverter();
+    }
+
+    @Bean("offsetTimeValueConverter")
+    ValueConverter offsetTimeValueConverter() {
+        return jsr310ConvertersConfiguration.offsetTimeValueConverter();
+    }
+
+    @Bean("offsetTimeStructuredBindingEditor")
+    TypedStructuredBindingEditor offsetTimeStructuredBindingEditor() {
+        return jsr310ConvertersConfiguration.offsetTimeStructuredBindingEditor();
+    }
+
+    @Bean("localDateTimeConverter")
+    FormattedValueConverter localDateTimeConverter() {
+        return jsr310ConvertersConfiguration.localDateTimeConverter();
+    }
+
+    @Bean("localDateTimeValueConverter")
+    ValueConverter localDateTimeValueConverter() {
+        return jsr310ConvertersConfiguration.localDateTimeValueConverter();
+    }
+
+    @Bean("localDateTimeStructuredBindingEditor")
+    TypedStructuredBindingEditor localDateTimeStructuredBindingEditor() {
+        return jsr310ConvertersConfiguration.localDateTimeStructuredBindingEditor();
+    }
+
+    @Bean("localDateConverter")
+    FormattedValueConverter localDateConverter() {
+        return jsr310ConvertersConfiguration.localDateConverter();
+    }
+
+    @Bean("localDateValueConverter")
+    ValueConverter localDateValueConverter() {
+        return jsr310ConvertersConfiguration.localDateValueConverter();
+    }
+
+    @Bean("localDateStructuredBindingEditor")
+    TypedStructuredBindingEditor localDateStructuredBindingEditor() {
+        return jsr310ConvertersConfiguration.localDateStructuredBindingEditor();
+    }
+
+    @Bean("localTimeConverter")
+    FormattedValueConverter localTimeConverter() {
+        return jsr310ConvertersConfiguration.localTimeConverter();
+    }
+
+    @Bean("localTimeValueConverter")
+    ValueConverter localTimeValueConverter() {
+        return jsr310ConvertersConfiguration.localTimeValueConverter();
+    }
+
+    @Bean("localTimeStructuredBindingEditor")
+    TypedStructuredBindingEditor localTimeStructuredBindingEditor() {
+        return jsr310ConvertersConfiguration.localTimeStructuredBindingEditor();
+    }
+
+    @Bean("zonedDateTimeConverter")
+    FormattedValueConverter zonedDateTimeConverter() {
+        return jsr310ConvertersConfiguration.zonedDateTimeConverter();
+    }
+
+    @Bean("zonedDateTimeValueConverter")
+    ValueConverter zonedDateTimeValueConverter() {
+        return jsr310ConvertersConfiguration.zonedDateTimeValueConverter();
+    }
+
+    @Bean("zonedDateTimeStructuredBindingEditor")
+    TypedStructuredBindingEditor zonedDateTimeStructuredBindingEditor() {
+        return jsr310ConvertersConfiguration.zonedDateTimeStructuredBindingEditor();
+    }
+
+    @Bean("periodValueConverter")
+    ValueConverter periodValueConverter() {
+        return jsr310ConvertersConfiguration.periodValueConverter();
+    }
+
+    @Bean("instantValueConverter")
+    ValueConverter instantValueConverter() {
+        return jsr310ConvertersConfiguration.instantValueConverter();
+    }
+
+    @Bean("defaultUUIDConverter")
+    protected UUIDConverter defaultuuidConverter() {
+        return new UUIDConverter();
     }
 
     @Bean("defaultGrailsBigIntegerConverter")

